osu.Game.Rulesets.Taiko/UI/DrawableTaikoMascot.cs

```diff
@@ -0,0 +1,105 @@
+﻿// Copyright (c) ppy Pty Ltd <contact@ppy.sh>. Licensed under the MIT Licence.
+// See the LICENCE file in the repository root for full licence text.
+
+using osu.Framework.Allocation;
+using osu.Framework.Audio.Track;
+using osu.Framework.Bindables;
+using osu.Framework.Graphics;
+using osu.Framework.Graphics.Textures;
+using osu.Game.Beatmaps.ControlPoints;
+using osu.Game.Graphics.Containers;
+
+namespace osu.Game.Rulesets.Taiko.UI
+{
+    public sealed class DrawableTaikoMascot : BeatSyncedContainer
+    {
+        private static TaikoMascotTextureAnimation idleDrawable, clearDrawable, kiaiDrawable, failDrawable;
+        private EffectControlPoint lastEffectControlPoint;
+        private TaikoMascotAnimationState state;
+
+        public Bindable<TaikoMascotAnimationState> PlayfieldState;
+
+        /// <summary>
+        /// Determines if there should be no "state logic", intended for testing.
+        /// </summary>
+        public bool Dumb { get; set; }
+
+        public TaikoMascotAnimationState State
+        {
+            get => state;
+            set
+            {
+                state = value;
+
+                foreach (var child in InternalChildren)
+                    child.Hide();
+
+                var drawable = getStateDrawable(State);
+
+                drawable?.Show();
+            }
+        }
+
+        public DrawableTaikoMascot(TaikoMascotAnimationState startingState = TaikoMascotAnimationState.Idle)
+        {
+            RelativeSizeAxes = Axes.Both;
+            PlayfieldState = new Bindable<TaikoMascotAnimationState>();
+            PlayfieldState.BindValueChanged(b =>
+            {
+                if (lastEffectControlPoint != null)
+                    State = getFinalAnimationState(lastEffectControlPoint, b.NewValue);
+            });
+
+            State = startingState;
+        }
+
+        private TaikoMascotTextureAnimation getStateDrawable(TaikoMascotAnimationState state)
+        {
+            return state switch
+            {
+                TaikoMascotAnimationState.Idle => idleDrawable,
+                TaikoMascotAnimationState.Clear => clearDrawable,
+                TaikoMascotAnimationState.Kiai => kiaiDrawable,
+                TaikoMascotAnimationState.Fail => failDrawable,
+                _ => null
+            };
```

Comment:
* We don't use `switch` expressions. Please use a normal `switch` instead.
* Not 100% sold on returning `null` instead of throwing, but I suppose in this case it's defensible.